Remove links

aydamacink opened the pull request:
I have some questions, regarding this pull request, some directly related to the code and some not. 
**Unrelated to the code**: As I was adding the files to be commited I noticed that Gemfile.lock and db/schema.rb had changed. However I didn't commit them.  In the Gemfile turbolinks and coffee rails were deleted & in the schema file  the time stamps were deleted and added again. Why does this happen and what is a good practice in these cases?

**Related to the code** :
1) I added a line ` helper_method :signed_in_as_admin? `within the ApplicationController .  But I think I could have also added the method to the  ApplicationHelper . 
How do you know what is best? and where should you add the method?

2) I could not test in the browser if this code really works since atm it is only possible to sign in as an admin. 

Thank you in advance 👍 💯

1000miles commented:
Hey @aydamacink, regarding your issue with the `Gemfile.lock`and `turbolinks`I can confirm that it also happend a few times in my branches. The `Gemfile.lock`was really stubborn and I could not easily check it out, so I had to remove it a few times and do a new `bundle install` again.

joecorcoran commented:
`Gemfile.lock` confused me recently, because no matter how many times I checked it out, it kept coming back! It turns out that `spring`, the tool that reloads the Rails app code in development, was running in the background and I had to run `spring stop` to actually be able to `git checkout Gemfile.lock`.

`Gemfile.lock` can be a problem for other reasons too. It sometimes changes because your version of Bundler is different to that of a colleague. I would suggest that you make sure you're up to date with master, run `bundle install` again, and then see if the changes in the `Gemfile.lock` file are real or not. Then you can decide if you should check it in.

`db/schema.rb` is a similar thing – sometimes it's only some timestamps that have changed on your machine and this is not really relevant for other people, so you have to look at the contents to decide. Timestamp changes are not worth checking in (IMO), but more significant changes like new or renamed tables, columns etc. definitely are!

> I added a line helper_method :signed_in_as_admin?within the ApplicationController . But I think I could have also added the method to the ApplicationHelper. How do you know what is best? and where should you add the method?

Using `helper_method` in the controller makes this method available in both controllers and views. If you need it in both places, that's a good way of doing it. If you define a method in `ApplicationHelper` it will never be available in controllers.

> I could not test in the browser if this code really works since atm it is only possible to sign in as an admin.

Good point! Let's fix that tonight.

aydamacink commented:
Last commit: Non admin users were unable to sign in, so we deleted the method that was in the SessionsController, but we left  the file because we the routes that we defined expect it to exist. 

We created a method called `disallow_unless_admin_or_user(user)`in the applications controller and then pass it in the before filter as a lambda, so it will be executed later when the user is already set. And we added `skip_before_action :redirect_signed_in_users, only: [:new, :create]` so the admin is not redirected after  creating a new user. We also removed this line `sign_in(@user)`because when admins create a user, we don't want the admin to be signed in as the user that was just created. 

We added the method `show_edit_links?(user)`to the ApplicationHelper, so we could use it in the view and now the admin, as well as the current_user can see the edit link. The admin for all the users and the current_user for her/his profile
